compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.h

```diff
@@ -117,12 +117,19 @@ SmallVector<Value> getTileSizes(OpBuilder &b, Operation *op, unsigned level);
 /// Sets the lowering configuration, overwriting existing attribute values.
 void setLoweringConfig(Operation *op, Attribute config);
 
+/// Sets an attribute to identify the rootOp and adds any information needed for
+/// the tuner from compiler. Currently, only sets a `UnitAttr`. Note that this
+/// attribute is not used by the compiler at any level and is only intended for
+/// tuner use.
+void setRootOpInfo(Operation *op);
```

Comment:
It is a deeply hidden warning to me. It could be outdated soon if other IREE contributors start relying on the attribute to do transformations. We could stop them, but it is like overheads++. Because reviewers need to know the information, and the contributors waste some time on the patch.

My expectation was guarding it over a flag in the first place, so it won't evolve without controls. If we start with guarding it over a flag, people who want to use the attribute will to be aware of this information.